sklearn/linear_model/base.py

```diff
@@ -32,7 +32,9 @@
 from ..utils.extmath import safe_sparse_dot
 from ..utils.sparsefuncs import mean_variance_axis, inplace_column_scale
 from ..utils.fixes import sparse_lsqr
-from ..utils.seq_dataset import ArrayDataset, CSRDataset
+from ..utils.seq_dataset import ArrayDataset32, CSRDataset32
+from ..utils.seq_dataset import ArrayDataset64 as ArrayDataset
```

Comment:
IMHO that's how I would do it for the moment. Like this you ensure that we don't break anything.